knowledge_verificator/command_line.py

```diff
@@ -79,11 +79,17 @@ def run_cli_mode():
                 if material is None:
                     continue
 
+                available_paragraphs: list[str] = [
+                    _paragraph
+                    for _paragraph in material.paragraphs
+                    if ac_module.choose_answer(_paragraph) is not None
+                ]
+
```

Comment:
I assume `ac_module.choose_answer` is expensive, just so cache the results and don't recheck all paragraphs every time.

Reply:
Good point, thanks Python developers for [functools](https://docs.python.org/3/library/functools.html) and `@lru_cache` and `@cache`.

Reply:
Or not as methods should not be cached (they cause the entire object to be stored in memory), I have implemented caching myself.